pyNN/nest/recording.py

```diff
@@ -422,14 +433,14 @@ def _reset(self):
         # Maybe we can reset them, rather than create new ones?
         self._multimeter = Multimeter()
         self._spike_detector = SpikeDetector()
-
-    def _get_spiketimes(self, id):
-        return self._spike_detector.get_spiketimes([id])[id]  # hugely inefficient - to be optimized later
-
+    
+    def _get_spiketimes(self, ids):
+        return self._spike_detector.get_spiketimes(ids)  # hugely inefficient - to be optimized later
```

Comment:
Yes, but since spikes are very sparse, it doesn't take too much time, at least in my use-cases compared to other operations, so I didn't go into optimizing it.